src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -367,4 +370,78 @@ public boolean contains(String key) {
 
         return mPrefs.contains(key);
     }
+    
+    private TokenCacheItem getFamilyRefreshTokenItemForUser(String userId) {       
+        
+        Iterator<TokenCacheItem> results = this.getTokensForUser(userId).iterator();
+        TokenCacheItem tokenItem = new TokenCacheItem();   
+        while (results.hasNext()) {
+            final TokenCacheItem tokenCacheItem = results.next();
+            if(tokenCacheItem.getUserInfo() != null
+                    && !StringExtensions.IsNullOrBlank(tokenCacheItem.getFamilyClientId())
+                    && tokenCacheItem.getFamilyClientId().equalsIgnoreCase("1")){
+                tokenItem = tokenCacheItem;
+                //return the cacheItem which contains the FRT for this userId
+                //there should be at most only one FRT record for each userId
+                return tokenItem;
+            }
+        }
+        //If the cache does not contain the FRT for this user return null
+        Logger.i(TAG, "Cannot find the family refresh token cache item for this user in cache", "");
+        return null;
+    }
+    
+    protected String serialize(String uniqueUserId) throws UsageAuthenticationException {
+        if (StringExtensions.IsNullOrBlank(uniqueUserId)) {
+            throw new IllegalArgumentException("uniqueUserId");
+        }
+        
+        if((new BrokerProxy(mContext)).canSwitchToBroker()){
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_EXPORT,"Failed to export the FID because broker is enabled.");
+        } else {
+            TokenCacheItem tokenItem = getFamilyRefreshTokenItemForUser(uniqueUserId);
+            
+            if(tokenItem == null) {
+                Logger.i(TAG, "Cannot serialize the FRT cache item: family refresh token is null", "");
+                return null;
+            }
+
+            Gson gson = new GsonBuilder()
+                    .registerTypeAdapter(BlobContainer.class, new BlobContainerAdapter())
```

Comment:
I am confused about your question here. What do you want to simplify here?  The class BlobContainerAdapter is to customize the serialize/deserialize in gson. Because we need the version UID serialized as well to achieve the backcompat. If we do not do this, there is no better way we could serialize the versionUID into the json string. We need versionUID to be checked when doing the deserialization.